google/resource_google_service_account_test.go

```diff
@@ -25,13 +27,26 @@ func TestAccGoogleServiceAccount_basic(t *testing.T) {
 				Config: testAccGoogleServiceAccountBasic(accountId, displayName),
 				Check: resource.ComposeTestCheckFunc(
 					testAccCheckGoogleServiceAccountExists("google_service_account.acceptance"),
+					resource.TestCheckResourceAttr(
+						"google_service_account.acceptance", "project", project),
 				),
 			},
 			// The second step updates the service account
 			resource.TestStep{
 				Config: testAccGoogleServiceAccountBasic(accountId, displayName2),
 				Check: resource.ComposeTestCheckFunc(
 					testAccCheckGoogleServiceAccountNameModified("google_service_account.acceptance", displayName2),
+					resource.TestCheckResourceAttr(
+						"google_service_account.acceptance", "project", project),
+				),
+			},
+			// The third step adds the default project to the service account
```

Comment:
Hmm, is this really what you want? Wouldn't this test just update the config to what it already is and then not actually do anything? I guess that's not a huge deal but if that's what you're going for it might be good to explain it in a comment. Otherwise maybe it deserves to be a separate TestCase, rather than a TestStep.

Reply:
Well, what I am trying to assert here is that adding the `project` field to an existing configuration with the same value as the default does not recreate the service account, which was not the case without the `Computed: true`.

But I realize that this test should rather assert that the service account's unique id does not change between step 2 and 3.

Reply:
Test case and comment updated, but not sure if this is the best way to compare state values between steps.

@danawillow, what do you think?